core/transaction/utils.go

```diff
@@ -0,0 +1,400 @@
+
+	rand := util.NewRand(numSharders)
+
+	selectedSharders := make([]string, 0, minNumConfirmation+1)
+
+	// random pick minNumConfirmation+1 first
+	for i := 0; i <= minNumConfirmation; i++ {
+		n, err := rand.Next()
+
+		if err != nil {
+			break
+		}
+
+		selectedSharders = append(selectedSharders, sharders[n])
+	}
+
+	numSuccess := 0
+
+	var retTxn *Transaction
+
+	//leave first item for ErrTooLessConfirmation
+	var msgList = make([]string, 1, numSharders)
+
+	urls := make([]string, 0, len(selectedSharders))
+
+	for _, sharder := range selectedSharders {
+		urls = append(urls, fmt.Sprintf("%v/%v%v", sharder, TXN_VERIFY_URL, txnHash))
+	}
+
+	header := map[string]string{
+		"Content-Type":                "application/json; charset=utf-8",
+		"Access-Control-Allow-Origin": "*",
+	}
+
+	transport := createTransport(resty.DefaultDialTimeout)
+
+	options := []resty.Option{
+		resty.WithTimeout(resty.DefaultRequestTimeout),
+		resty.WithRetry(resty.DefaultRetry),
+		resty.WithHeader(header),
+		resty.WithTransport(transport),
+	}
+
+	r := resty.New(options...).
+		Then(func(req *http.Request, resp *http.Response, respBody []byte, cf context.CancelFunc, err error) error {
+			url := req.URL.String()
+
+			if err != nil { //network issue
+				msgList = append(msgList, err.Error())
+				return err
+			}
+
+			if resp.StatusCode != 200 {
+				msgList = append(msgList, url+": ["+strconv.Itoa(resp.StatusCode)+"] "+string(respBody))
+				return errors.Throw(ErrInvalidRequest, strconv.Itoa(resp.StatusCode)+": "+resp.Status)
+			}
+
+			var objmap map[string]json.RawMessage
+			err = json.Unmarshal(respBody, &objmap)
+			if err != nil {
+				msgList = append(msgList, "json: "+string(respBody))
+				return err
+			}
+			txnRawJSON, ok := objmap["txn"]
+
+			// txn data is found, success
+			if ok {
+				txn := &Transaction{}
+				err = json.Unmarshal(txnRawJSON, txn)
+				if err != nil {
+					msgList = append(msgList, "json: "+string(txnRawJSON))
+					return err
+				}
+				if len(txn.Signature) > 0 {
+					retTxn = txn
+				}
+				numSuccess++
+
+			} else {
+				// txn data is not found, but get block_hash, success
+				if _, ok := objmap["block_hash"]; ok {
+					numSuccess++
+				} else {
+					// txn and block_hash
+					msgList = append(msgList, fmt.Sprintf("Sharder does not have the block summary with url: %s, contents: %s", url, string(respBody)))
+				}
+
+			}
+
+			return nil
+		})
+
+	for {
+		r.DoGet(context.TODO(), urls...)
+
+		r.Wait()
+
+		if numSuccess >= minNumConfirmation {
+			break
+		}
+
+		// pick more one sharder to query transaction
+		n, err := rand.Next()
+
+		if errors.Is(err, util.ErrNoItem) {
+			break
+		}
+
+		urls = []string{fmt.Sprintf("%v/%v%v", sharders[n], TXN_VERIFY_URL, txnHash)}
+
+	}
+
+	if numSuccess > 0 && numSuccess >= minNumConfirmation {
+		if retTxn == nil {
+			return nil, errors.Throw(ErrNoTxnDetail, strings.Join(msgList, "\r\n"))
```

Comment:
msgList[0] is not set in this case, so perhaps `msgList[1:]`. Or just define the msgList normally as `msgList := make([]string, 0, numSharders)` and append it to TooLessConfirmation below as `append(TooLessConfirmErr, msgList...)`.

Reply:
this code is moved from and is not changed, so this might be not related to this issue, let's do it in different issue
```
func VerifyTransaction(txnHash string, sharders []string) (*Transaction, error) {
```